homeassistant/components/evohome/__init__.py

```diff
@@ -486,24 +487,28 @@ async def _update_v2(self, *args, **kwargs) -> None:
         except (aiohttp.ClientError, evohomeasync2.AuthenticationError) as err:
             _handle_exception(err)
         else:
-            async_dispatcher_send(self.hass, DOMAIN)
-
             _LOGGER.debug("Status = %s", status)
 
         if access_token != self.client.access_token:
             await self.save_auth_tokens()
 
-    async def async_update(self, *args, **kwargs) -> None:
+    async def async_update(self, *args, manual_update=None, **kwargs) -> None:
         """Get the latest state data of an entire Honeywell TCC Location.
 
         This includes state data for a Controller and all its child devices, such as the
         operating mode of the Controller and the current temp of its children (e.g.
         Zones, DHW controller).
+
+        The state is updated periodically. It is also updated whenever a client API has
+        been invoked, after waiting a moment for things to quiesce (manual_update=True).
         """
-        await self._update_v2()
+        if manual_update:
+            await asyncio.sleep(1)
+
+        elif self.client_v1:  # invoking a client API won't affect temps
+            await self._update_v1_api_state()
 
-        if self.client_v1:
-            await self._update_v1()
+        await self._update_v2_api_state()
```

Comment:
In the end, a much bigger change than I'd hoped, but pounced on a few minor bugs, and better explicit/implicit documentation. I hope this is OK.
 - calling `async_dispatcher_send(self.hass, DOMAIN)` twice instead of once
 - calling `self._update_v1_api_state()` when not required

Reply:
It feels all a little chaotic with all those keywords to change execution. Instead of passing a keyword, we should do a new function.

Reply:
OK, best of both worlds - removed the `manual_update` keyword, but didn't have to add a new function.